build/DownloadCFamilyPlugin/DownloadAndExtractSonarJS.cs

```diff
@@ -0,0 +1,119 @@
+        //          package\bin\
+        //          package\lib\
+        //          package\node_modules\
+        //          package\package.json
+
+        // The archive containing the eslintbridge
+        private const string SourceEsLintBridgeFilePattern = "eslint-bridge-*.tgz";
+
+        // Sub-folder into which the tar file should be unzipped
+        public const string TargetEslintBridgeFolderName = "eslint-bridge";
+
+        // List of patterns to match single files in the uncompressed output
+        private readonly string[] SourceSingleFilePatterns = new string[]
+        {
+                "Sonar_way_profile.json",
+                "Sonar_way_recommended_profile.json"
+        };
+
+        private readonly string SourceRelativePackageDirectory = TargetEslintBridgeFolderName + @"\package";
+
+        #region MSBuild input / output properties (set/read in the project file)
+
+        // Download url example: https://binaries.sonarsource.com/Distribution/sonar-javascript-plugin/sonar-javascript-plugin-6.2.0.12043.jar
+        [Required]
+        public string DownloadUrl { get; set; }
+
+        [Output]
+        public string[] FilesToEmbed { get; private set; }
+
+        [Output]
+        public string PackageDirectoryToEmbed { get; private set; }
+
+        #endregion
+
+        public override bool Execute()
+        {
+            Log.LogMessage(MessageImportance.High, $"Download url: {DownloadUrl}");
+
+            var pluginFileName = Common.ExtractPluginFileNameFromUrl(DownloadUrl, Log);
+
+            // Ensure working directories exists
+            var localWorkingFolder = Path.Combine(Environment.GetEnvironmentVariable("LocalAppData"), "SLVS_TypeScript_Build");
+            var perVersionPluginFolder = Path.Combine(localWorkingFolder, Path.GetFileNameWithoutExtension(pluginFileName));
+            Common.EnsureWorkingDirectoryExist(perVersionPluginFolder, this.Log);
+
+            // Download and unzip the jar
+            var jarFilePath = Path.Combine(perVersionPluginFolder, pluginFileName);
+            Common.DownloadJarFile(DownloadUrl, jarFilePath, Log);
+            Common.UnzipJar(jarFilePath, perVersionPluginFolder, Log);
+
+            // Uncompress and extract the windows tar archive to get the eslint-bridge folder
+            var tarFilePath = Common.FindSingleFile(perVersionPluginFolder, SourceEsLintBridgeFilePattern, Log);
+            var tarSubFolder = Path.Combine(perVersionPluginFolder, TargetEslintBridgeFolderName);
+            Common.UncompressAndUnzipTgz(tarFilePath, tarSubFolder, Log);
+
+            // Locate the required files from the uncompressed jar and tar
+            var fileList = FindFiles(perVersionPluginFolder);
+            FilesToEmbed = fileList.ToArray();
+
+            PackageDirectoryToEmbed =   Path.Combine(perVersionPluginFolder, SourceRelativePackageDirectory);
+
+            return !Log.HasLoggedErrors;
```

Comment:
This code is logically the same as the CFamily code. The differences are due to:
* different compression formats used,
* different internal layouts in the archive, and
* the file patterns we want to extract.